### PR TITLE
Add reload command, config update, change config on damage listener

### PR DESCRIPTION
**Pull request by dhannyjsb**

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -8,19 +8,29 @@
...
   <properties>
     <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
     <maven.compiler.source>11</maven.compiler.source>
     <maven.compiler.target>11</maven.compiler.target>
   </properties>
   <dependencies>
     <dependency>
-      <groupId>io.papermc.paper</groupId>
-      <artifactId>paper</artifactId>
+      <groupId>org.spigotmc</groupId>
```
Comment (MagicCheese1) [inline pom.xml:24]: Please keep paper for now so it works with the current github workflow

**Review by MagicCheese1 — CHANGES_REQUESTED** [review]
minor changes

File: pom.xml (hunk 1 above)
Comment (dhannyjsb) [reply pom.xml]: Okay. I've changed it

File: src/main/java/com/github/magiccheese1/damageindicator/BukkitEventListener.java [hunk 2]
```diff
@@ -43,7 +41,7 @@ public void entityDamageByEntity(EntityDamageByEntityEvent event) {
         Location spawnLocation;
         Random random = new Random();
         DecimalFormat damageFormat = new DecimalFormat(
-                ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(config.getString("IndicatorFormat"))));
+                ChatColor.translateAlternateColorCodes('&', Objects.requireNonNull(this.plugin.getConfig().getString(Utility.FORMAT_INDICATOR))));
```
[the remark this answers is not in the document]
Comment (dhannyjsb) [reply src/main/java/com/github/magiccheese1/damageindicator/BukkitEventListener.java]: Sure, can you check new commit.

File: src/main/java/com/github/magiccheese1/damageindicator/Cmd.java [hunk 3]
```diff
@@ -0,0 +1,28 @@
+package com.github.magiccheese1.damageindicator;
+
+import org.bukkit.ChatColor;
+import org.bukkit.command.Command;
+import org.bukkit.command.CommandExecutor;
+import org.bukkit.command.CommandSender;
+
+public class Cmd implements CommandExecutor {
```
[the remark this answers is not in the document]
Comment (dhannyjsb) [reply src/main/java/com/github/magiccheese1/damageindicator/Cmd.java]: I've changed this class name.

**Comment by MagicCheese1** [comment]
When i build it and try to run it, I get a noClassDefFoundError for ConfigUpdater. Am I doing something wrong? Have you even tested this?

File: src/main/java/com/github/magiccheese1/damageindicator/Main.java [hunk 4]
```diff
@@ -3,21 +3,32 @@
...
         // Save the default config from src/resources/config.yml
-        this.saveDefaultConfig();
-        // Get current config
-        FileConfiguration config = this.getConfig();
-
+        saveDefaultConfig();
+        // The config needs to exist before using the updater
+        File configFile = new File(getDataFolder(), "config.yml");
+        try {
+            ConfigUpdater.update(this, "config.yml", configFile, Arrays.asList("..."));
```
Comment (lynxplay) [inline src/main/java/com/github/magiccheese1/damageindicator/Main.java:25]: What is `"..."` supposed to do here ?

Comment (dhannyjsb) [reply src/main/java/com/github/magiccheese1/damageindicator/Main.java:25]: `"..."` indicate there's no array on config.yml

Comment (lynxplay) [reply src/main/java/com/github/magiccheese1/damageindicator/Main.java:25]: If I understand the ConfigUpdater correctly this argument ([see here](https://github.com/tchristofferson/Config-Updater/blob/master/src/main/java/com/tchristofferson/configupdater/ConfigUpdater.java#L33)) represents the ignore sections that will not be updated.

If you want to indicate that no section should have this treatment, an empty list using [Collections.emptyList()](https://docs.oracle.com/en/java/javase/16/docs/api/java.base/java/util/Collections.html#emptyList()) might be a better solution.

Comment (dhannyjsb) [reply src/main/java/com/github/magiccheese1/damageindicator/Main.java:25]: Sure. thank you for your solution.

**Comment by MagicCheese1** [comment]
> > When i build it and try to run it, I get a noClassDefFoundError for ConfigUpdater. Am I doing something wrong? Have you even tested this?
> 
> yes, I have test before pull, ConfigUpdater shade from https://github.com/tchristofferson/Config-Updater

Can you include maven configuration for shading that?

**Comment by dhannyjsb** [comment]
> 
> 
> > > When i build it and try to run it, I get a noClassDefFoundError for ConfigUpdater. Am I doing something wrong? Have you even tested this?
> > 
> > 
> > yes, I have test before pull, ConfigUpdater shade from https://github.com/tchristofferson/Config-Updater
> 
> Can you include maven configuration for shading that?



> 
> 
> > > When i build it and try to run it, I get a noClassDefFoundError for ConfigUpdater. Am I doing something wrong? Have you even tested this?
> > 
> > 
> > yes, I have test before pull, ConfigUpdater shade from https://github.com/tchristofferson/Config-Updater
> 
> Can you include maven configuration for shading that?

Okay, can you review my latest commit from pom.xml

**Review by MagicCheese1 — APPROVED** [review]
Everything looks good. Thanks for the pr

**Pull request merged**